controllers/serviceaccount/serviceaccount_remover.go

```diff
@@ -0,0 +1,109 @@
+package serviceaccount
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudogu/cesapp/v4/core"
+	"github.com/cloudogu/cesapp/v4/registry"
+	k8sv1 "github.com/cloudogu/k8s-dogu-operator/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Remover is the unit to handle the removal of service accounts
+type Remover struct {
+	Registry registry.Registry `json:"registry"`
+	Executor CommandExecutor   `json:"executor"`
+}
+
+// NewServiceAccountRemover creates a new instance of ServiceAccountCreator
+func NewServiceAccountRemover(registry registry.Registry, commandExecutor CommandExecutor) *Remover {
+	return &Remover{
+		Registry: registry,
+		Executor: commandExecutor,
+	}
+}
+
+// RemoveServiceAccounts removes all service accounts for a given dogu
+func (r *Remover) RemoveServiceAccounts(ctx context.Context, doguResource *k8sv1.Dogu, dogu *core.Dogu) error {
+	logger := log.FromContext(ctx)
+
+	for _, serviceAccount := range dogu.ServiceAccounts {
+		parentPath := "sa-" + serviceAccount.Type
+		doguConfig := r.Registry.DoguConfig(dogu.GetSimpleName())
+
+		// check if the service account really exists
+		exists, err := serviceAccountExists(parentPath, doguConfig)
+		if err != nil {
+			return err
+		}
+
+		if !exists {
+			logger.Info("skipping removal of service account because it does not exists")
+			continue
+		}
+
+		// check if the dogu is enabled
+		doguRegistry := r.Registry.DoguRegistry()
+		enabled, err := doguRegistry.IsEnabled(serviceAccount.Type)
+		if err != nil {
+			return fmt.Errorf("failed to check if dogu %s is enabled: %w", serviceAccount.Type, err)
+		}
+		if !enabled {
+			logger.Info("skipping removal of service account because dogu is not enabled")
+			continue
+		}
+
+		// get the service account dogu.json
+		targetDescriptor, err := doguRegistry.Get(serviceAccount.Type)
+		if err != nil {
+			return fmt.Errorf("failed to get service account dogu.json: %w", err)
+		}
+
+		// execute remove service account command
+		err = r.executeServiceAccountRemoveCommand(ctx, dogu, targetDescriptor, doguResource.Namespace, serviceAccount)
+		if err != nil {
+			return fmt.Errorf("failed to execute service account remove command: %w", err)
+		}
+
+		// remove credentials
+		err = doguConfig.DeleteRecursive(parentPath)
+		if err != nil {
+			return fmt.Errorf("failed to remove service account from config: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (r *Remover) executeServiceAccountRemoveCommand(ctx context.Context, consumerDogu *core.Dogu, saDogu *core.Dogu, namespace string, serviceAccount core.ServiceAccount) error {
+	removeCommand := r.getRemoveCommand(saDogu)
+	if removeCommand == nil {
+		return fmt.Errorf("service account dogu does not expose remove command")
```

Comment:
I suggest to add a bit context to the error message unless later the context is added, f. i. which dogu is affected here?

Also I suggest to move the error checking into the Command finding method because they are part of the same routine. It is quite idiomatic to find a needle in a haystack and throw an error when the needle is not found.